lib/main.dart

```diff
@@ -41,23 +42,39 @@ class MyApp extends StatelessWidget {
         ChangeNotifierProvider(create: (_) => ScheduleLayoutSettingProvider()),
         ChangeNotifierProvider(create: (_) => SavedSubjectsProvider()),
       ],
-      child: MaterialApp(
-        title: 'IIUM Schedule',
-        theme: ThemeData(primarySwatch: Colors.purple),
-        darkTheme: ThemeData.dark().copyWith(
-          cupertinoOverrideTheme:
-              const CupertinoThemeData(primaryColor: Colors.purple),
-          textButtonTheme: TextButtonThemeData(
-            style:
-                TextButton.styleFrom(foregroundColor: Colors.purple.shade200),
-          ),
-          outlinedButtonTheme: OutlinedButtonThemeData(
-            style: OutlinedButton.styleFrom(
-                foregroundColor: Colors.purple.shade200),
-          ),
-        ),
-        themeMode: ThemeMode.system,
-        home: const MyBody(),
+      child: DynamicColorBuilder(
+        builder: (ColorScheme? lightColorScheme, ColorScheme? darkColorScheme) {
+          return MaterialApp(
+            title: 'IIUM Schedule',
+            theme: ThemeData(
+              colorScheme: lightColorScheme,
```

Comment:
```suggestion
              colorScheme: lightColorScheme ?? const ColorScheme.light(),
```

However, it is not a one liner fix. but just enough as a fallback colour scheme.